magento/pwa-studio
# [PWA-335] Ensure apollo cache is updated after mutations/queries

## Description

Right now we have part of the app making mutations or queries to graphql without any knowledge of the other parts of the app. These should all be modified to use named fragments to prevent over fetch. For example, {{addItemToCart}}/{{removeItemFromCart}} could use a {{CartTriggerFragment}} and should definitely be using the {{CartPageFragment}} so that the cart trigger/page is always up to date. Right now the trigger has to manually refetch details and without using the cart page fragment the cart page won't be updated to reflect the proper state of the cart after the mutation.

So, where applicable, replace refetchQueries and incomplete query response bodies in product/cart mutations/queries with named fragments such as {{CartPageFragment}} or {{CartTriggerFragment}}.

### Development Note
The lines between another very similar task [PWA-321](https://jira.corp.magento.com/browse/PWA-321) were hard to align with, so most of the scope of that task has crept into this one a bit. It should work now that CartPage should be able to solely rely on whats in the cache, regardless of where the mutation occurred. 

## Related Issue





* [[PWA-335](https://jira.corp.magento.com/browse/PWA-335)] Ensure apollo cache is updated after mutations/queries

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Add Products of every type to your cart
2. Verify that the cart counter maintains the correct number of items
3. Navigate to `/cart` and verify all data is displayed, no loading indicator was rendered
4. Modify a product quantity, verify cart counter adjusts correctly
5. Modify a configurable item from MiniCart (options + quantity)
6. Verify cart counter changed and cart page rendered change w/o having to refresh

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
